crates/extrinsics/src/balance.rs

```diff
@@ -35,10 +38,13 @@ use anyhow::{
     Context,
     Result,
 };
+use url::Url;
+
+use crate::url_to_string;
 
 /// Represents different formats of a balance
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub enum BalanceVariant<Balance: Clone> {
+pub enum BalanceVariant<Balance> {
```

Comment:
I fixed the description. I wanted to avoid adding restrictions such as From<u128> or Into<u128>. Additionally, I do not see any benefits in holding BalanceVariant in the extrinsic crate, which increases the code complexity. Instead, I decided to use it as a balance converter in cargo-contract crate before creating extrinsic types.